client/core/core.go

```diff
@@ -5118,6 +5111,49 @@ func processPreimageRequest(c *Core, dc *dexConnection, reqID uint64, oid order.
 	return nil
 }
 
+func acceptCsum(dc *dexConnection, reqID uint64, tracker *trackedTrade, isCancel bool, commitChecksum dex.Bytes) error {
+	checkCsum := func(csum dex.Bytes) error {
+		if csum != nil {
+			csumErr := errors.New("csum is already initialized")
+			resp, err := msgjson.NewResponse(reqID, nil,
+				msgjson.NewError(msgjson.InvalidRequestError, csumErr.Error()))
+			if err != nil {
+				return fmt.Errorf("preimage response encoding error: %w", err)
+			}
+			err = dc.Send(resp)
+			if err != nil {
+				return fmt.Errorf("preimage send error: %w", err)
+			}
+			return csumErr
+		}
+		return nil
+	}
+
+	tracker.mtx.Lock()
+	defer tracker.mtx.Unlock()
```

Comment:
Just want to point out that `err = dc.Send(resp)` (which currently has a timeout of `writeWait = time.Second * 3`) might happen when `mtx` is locked, it is usually a bad practice to hold `mtx` for too long (doing network calls and such), 

so this might be an attack vector for a `server` to cause a DOS to `client`, any thoughts on that ?

Reply:
I rewrote the implementation a bit to solve this issue.

Reply:
Thanks, I would have commented on that.  Def want to avoid high latency ops with this mutex held.  Will review your latest solution shortly.